pkg/drain/nodegroup.go

```diff
@@ -127,35 +136,58 @@ func (n *NodeGroupDrainer) Drain() error {
 			}
 
 			if newPendingNodes.Len() == 0 {
-				logger.Success("drained all nodes: %v", drainedNodes.List())
+				waitForAllRoutinesToFinish(ctx, sem, parallelLimit)
+				logger.Success("drained all nodes: %v", mapToList(drainedNodes.Items()))
 				return nil // no new nodes were seen
 			}
 
-			logger.Debug("already drained: %v", drainedNodes.List())
+			logger.Debug("already drained: %v", mapToList(drainedNodes.Items()))
 			logger.Debug("will drain: %v", newPendingNodes.List())
-
 			for i, node := range newPendingNodes.List() {
-				pending, err := n.evictPods(node)
-				if err != nil {
-					logger.Warning("pod eviction error (%q) on node %s", err, node)
-					time.Sleep(retryDelay)
-					continue
-				}
-				logger.Debug("%d pods to be evicted from %s", pending, node)
-				if pending == 0 {
-					drainedNodes.Insert(node)
+				if err := sem.Acquire(ctx, 1); err != nil {
+					logger.Critical("failed to claim sem: %w", err)
 				}
 
-				// only wait if we're not on the last node of this iteration
-				if n.nodeDrainWaitPeriod > 0 && i < newPendingNodes.Len()-1 {
-					logger.Debug("waiting for %.0f seconds before draining next node", n.nodeDrainWaitPeriod.Seconds())
-					time.Sleep(n.nodeDrainWaitPeriod)
-				}
+				go func(i int, node string) {
+					defer sem.Release(1)
+					logger.Debug("starting drain of node %s", node)
+					pending, err := n.evictPods(node)
+					if err != nil {
+						logger.Warning("pod eviction error (%q) on node %s", err, node)
+						time.Sleep(retryDelay)
+						return
+					}
+
+					logger.Debug("%d pods to be evicted from %s", pending, node)
+					if pending == 0 {
+						drainedNodes.Set(node, nil)
+					}
+
+					if n.nodeDrainWaitPeriod > 0 {
+						logger.Debug("waiting for %.0f seconds before draining next node", n.nodeDrainWaitPeriod.Seconds())
+						time.Sleep(n.nodeDrainWaitPeriod)
+					}
+				}(i, node)
 			}
 		}
 	}
 }
 
+func waitForAllRoutinesToFinish(ctx context.Context, sem *semaphore.Weighted, size int64) {
+	if err := sem.Acquire(ctx, size); err != nil {
+		logger.Critical("failed to claim sem: %w", err)
```

Comment:
```suggestion
		logger.Critical("failed to acquire semaphore while waiting for all routines to finish: %w", err)
```

Just so we know which one of the two failed to acquire a semaphore. :)